llvm/include/llvm/Transforms/Utils/SampleProfileLoaderBaseImpl.h

```diff
@@ -125,17 +125,15 @@ class PseudoProbeManager {
   }
 
   bool profileIsValid(const Function &F, const FunctionSamples &Samples) const {
-    const auto *Desc = getDesc(F);
-    if (!Desc) {
-      LLVM_DEBUG(dbgs() << "Probe descriptor missing for Function "
-                        << F.getName() << "\n");
+    // The desc for import function is unavailable. Check the function attribute
+    // for mismatch.
+    if (F.hasFnAttribute("profile-checksum-mismatch"))
```

Comment:
Should we check attribute only when descriptor isn't available? that's what the comment says..

Reply:
And when both desc and attribute are available, maybe assert they are consistent?

Reply:
Good point, done.